addons/mail/models/res_partner.py

```diff
@@ -261,13 +261,24 @@ def _get_mention_suggestions_domain(self, search):
         ])
 
     @api.model
-    def _search_mention_suggestions(self, domain, limit):
-        domain_is_user = expression.AND([[('user_ids', '!=', False)], [('user_ids.active', '=', True)], domain])
+    def _search_mention_suggestions(self, domain, limit, channel=None):
+        base_domain = expression.AND([[('user_ids', '!=', False)], [('user_ids.active', '=', True)]])
+        domain_is_user = expression.AND([base_domain, domain])
         priority_conditions = [
             expression.AND([domain_is_user, [('partner_share', '=', False)]]),  # Search partners that are internal users
             domain_is_user,  # Search partners that are users
             domain,  # Search partners that are not users
         ]
+        if channel:
```

Comment:
Channel makes no sense in mail, it should be done with an override in discuss directory. I would advice adding a "extra_domain" parameter that caller will use accordingly.

Reply:
> Thanks for comment, can we do this instead! since discuss models are loading before this file.
```py
 @api.model
    def _search_mention_suggestions(self, domain, limit, extra_domain=None):
        domain_is_user = expression.AND([[('user_ids', '!=', False)], [('user_ids.active', '=', True)], domain])
        priority_conditions = [
            expression.AND([domain_is_user, [('partner_share', '=', False)]]),  # Search partners that are internal users
            domain_is_user,  # Search partners that are users
            domain,  # Search partners that are not users
        ]
        if extra_domain:
            priority_conditions.append(extra_domain)
        .
        .
```